docs/source/user-guide/plugins.md

```diff
@@ -312,9 +312,18 @@ class TutorialDataset(Plugin):
         return None
 ```
 
+```{note}
+Some routers may want to cache data computed from datasets that they serve to avoid unnecessary recomputation. In this case, routers may rely on the
+`_xpublish_id` attribute (`DATASET_ID_ATTR_KEY` from `xpublish.api`) on each dataset. If this attribute is set, it should be a unique identifier for the dataset, otherwise the `dataset_id` used to load the dataset will be set as the `_xpublish_id` automatically.
+```
+
```

Comment:
It's probably worth being more explicit, and making it more important than alert. Also could you tweak the example above to explicit set the attr as an example?

Reply:
Do you mean move that up and change it to a warning? Ill update the example code too

Reply:
Ya, warning is probably the right level. I was thinking tweaking the tone as well, 'you need to set a unique`DATASET_ID_ATTR_KEY` from `xpublish.api` on each dataset for routers to manage caching appropriately' to go with the warning.

Reply:
Nice. I like that, K updated it!